### PR TITLE
[BEAM-2810] fastavro integration test

### PR DESCRIPTION
**Pull request by ryan-williams**
Add an integration test of the new fastavro support in the Python SDK's Avro IO

This was pulled out of #5496; some perf analysis is [here](https://gist.github.com/ryan-williams/ede5ae61605e7ba6aa655071858ef52b).

@chamikaramj has state on this, but is out for a few weeks; @herohde may be a good person to have a look? Thanks!

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | --- | --- | --- | ---

### CODE REVIEW

**Comment by herohde** [comment]
I'm not familiar with the Python SDK. @altay ?

File: sdks/python/apache_beam/examples/fastavro_it_test.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+    # pylint: disable=expression-not-assigned
+    {
+        'avro': avro_records,
+        'fastavro': fastavro_records
+    } \
+    | CoGroupByKey() \
+    | Map(check)
+
+    fastavro_read_pipeline.run().wait_until_finish()
+
```
Comment (aaltay) [inline sdks/python/apache_beam/examples/fastavro_it_test.py]: Do you also need to have equivalent of `assert result.state == PipelineState.DONE` here?

**Comment by aaltay** [comment]
Run Python PostCommit

**Comment by ryan-williams** [comment]
Run Python PostCommit

File: sdks/python/apache_beam/examples/fastavro_it_test.py (hunk 1 above)
Comment (ryan-williams) [reply sdks/python/apache_beam/examples/fastavro_it_test.py]: good point; added!

**Comment by ryan-williams** [comment]
The previous PostCommit run had my fastavro IT time out, so I dropped the default number of records from 10MM to 1MM, and now it passes. 

That seems fine to me, though it would be nice if we could see/track the timings of the {avro,fastavro} stages, to verify/demonstrate the speed difference… not sure if the IT machinery gives us that for free or whether I'd have to do something else to get it.

Either way, having it verify correctness is a win; thanks for the review!

**Comment by aaltay** [comment]
Thank you I will merge it.

I do not think our our IT infra gives us the stage timing information for free. It might be possible to do this with Dataflow per-stage counters.

**Pull request merged**